src/python/bcc/table.py

```diff
@@ -430,11 +430,11 @@ def __setitem__(self, key, leaf):
 
     def __delitem__(self, key):
         key = self._normalize_key(key)
-        # Deleting from array type maps does not have an effect, so
-        # zero out the entry instead.
+        super(ArrayBase, self).__delitem__(key)
+
+    def clearitem(self, key):
         leaf = self.Leaf()
```

Comment:
should we do _normalize_key here?

Reply:
Yes, The user's of `ArrayBase`'s __delitem__ after this patch are, `PerfEventArray` and `ProgArray`. Both these were calling `_normalize_key` in the previous code (before this patch). 

However I screwed up the normalization for __delitem__ in `PerCpuArray` and `Array`. Those need to normalize too before calling `clearitem`. Or `clear_item` itself should normalize within. I'll fix that in the next rev. thanks

Reply:
Fixed this.